### ISSUE TITLE
Restore failed | OSX 10.11.6 | LeetDown 1.0.5

### ISSUE THREAD

**Issue by PENG-OX** [rA9stuff/LeetDown#63]
2021-07-23 10:07:28.747 leetdown[43968:216087] Failed to connect (ipswselectstring) outlet from (LeetDown.ViewController) to (NSButton): missing setter or instance variable
2021-07-23 10:07:28.747 leetdown[43968:216087] Failed to connect (leetdownlabel) outlet from (LeetDown.ViewController) to (NSTextField): missing setter or instance variable
rm: no_device: No such file or directory
rm: no_dfu: No such file or directory
rm: no_dfu: No such file or directory
rm: no_device: No such file or directory
rm: libusb_missing: No such file or directory
rm: *.im4p: No such file or directory
rm: *.stock: No such file or directory
rm: *.patched: No such file or directory
rm: *.shsh: No such file or directory
rm: *.shsh2: No such file or directory
rm: *.bbfw: No such file or directory
rm: no_ipsw: No such file or directory
rm: no_dfu: No such file or directory
rm: no_device: No such file or directory
Your current ApNonce is ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997
7758956dff0
Your device is an iPad4,4
Your boardconfig is j85ap
rm: no_device: No such file or directory
rm: no_dfu: No such file or directory
rm: libusb_missing: No such file or directory
libusb exists.
No matching processes belonging to you were found
Device entered pwnedDFU successfully
No matching processes belonging to you were found
No matching processes belonging to you were found
Device is already in pwnedDFU
No matching processes belonging to you were found
Your current ApNonce is ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997
7758956dff0
Your device is an iPad4,4
Your boardconfig is j85ap
Version: c0b554e83f54d39d90cac9791160bf2ccb062aed - 355
[TSSC] manually specified ECID to use, parsed "7758956dff0" to dec:8201396740080 hex:7758956dff0
[TSSC] manually specified ApNonce to use, parsed "ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997" to hex:ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997
[TSSC] opening BM/iPad4,4.plist
[TSSR] LOG: device iPad4,4 doesn't need a baseband ticket, continuing without requesting a Baseband ticket
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... success
also requesting APTicket for update installing
[Error] [TSSR] Error: could not get id0 for installType=Update
[WARNING] [TSSR] faild to build tssrequest for alternative installType
[TSSR] User specified not to request a baseband ticket.
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... failure
[Error] ERROR: TSS request failed (status=128, message=An internal error occurred.)
Saved signing tickets!

iOS 10.3.3 for device iPad4,4 IS being signed!
Your current ApNonce is ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997
7758956dff0
Your device is an iPad4,4
Your boardconfig is j85ap
Archive:  ../stock.ipsw
  inflating: 058-74940-063.dmg       
  inflating: 058-75094-062.dmg       
  inflating: ._058-75094-062.dmg     
  inflating: 058-75381-062.dmg       
  inflating: BuildManifest.plist     
   creating: Firmware/
   creating: Firmware/all_flash/
  inflating: Firmware/all_flash/applelogo@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/applelogo@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/batterycharging0@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/batterycharging0@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/batterycharging1@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/batterycharging1@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/batteryfull@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/batteryfull@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/batterylow0@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/batterylow0@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/batterylow1@2x~ipad.s5l8960x.im4p  
  inflating: Firmware/all_flash/batterylow1@2x~ipad.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j71ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j71ap.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j72ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j72ap.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j73ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j73ap.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j85ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j85ap.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j86ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j86ap.im4p.plist  
  inflating: Firmware/all_flash/DeviceTree.j87ap.im4p  
  inflating: Firmware/all_flash/DeviceTree.j87ap.im4p.plist  
  inflating: Firmware/all_flash/glyphplugin@2x~ipad-lightning.s5l8960x.im4p  
  inflating: Firmware/all_flash/glyphplugin@2x~ipad-lightning.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/iBoot.ipad4.RELEASE.im4p  
  inflating: Firmware/all_flash/iBoot.ipad4.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/iBoot.ipad4b.RELEASE.im4p  
  inflating: Firmware/all_flash/iBoot.ipad4b.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/LLB.ipad4.RELEASE.im4p  
  inflating: Firmware/all_flash/LLB.ipad4.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/LLB.ipad4b.RELEASE.im4p  
  inflating: Firmware/all_flash/LLB.ipad4b.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/recoverymode@2x~ipad-lightning.s5l8960x.im4p  
  inflating: Firmware/all_flash/recoverymode@2x~ipad-lightning.s5l8960x.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j71.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j71.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j72.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j72.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j73.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j73.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j85.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j85.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j86.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j86.RELEASE.im4p.plist  
  inflating: Firmware/all_flash/sep-firmware.j87.RELEASE.im4p  
  inflating: Firmware/all_flash/sep-firmware.j87.RELEASE.im4p.plist  
   creating: Firmware/dfu/
  inflating: Firmware/dfu/iBEC.ipad4.RELEASE.im4p  
  inflating: Firmware/dfu/iBEC.ipad4.RELEASE.im4p.plist  
  inflating: Firmware/dfu/iBEC.ipad4b.RELEASE.im4p  
  inflating: Firmware/dfu/iBEC.ipad4b.RELEASE.im4p.plist  
  inflating: Firmware/dfu/iBSS.ipad4.RELEASE.im4p  
  inflating: Firmware/dfu/iBSS.ipad4.RELEASE.im4p.plist  
  inflating: Firmware/dfu/iBSS.ipad4b.RELEASE.im4p  
  inflating: Firmware/dfu/iBSS.ipad4b.RELEASE.im4p.plist  
  inflating: Firmware/Mav7Mav8-7.60.00.Release.bbfw  
  inflating: Firmware/Mav7Mav8-7.60.00.Release.plist  
   creating: Firmware/usr/
   creating: Firmware/usr/local/
   creating: Firmware/usr/local/standalone/
  inflating: kernelcache.release.ipad4  
  inflating: kernelcache.release.ipad4b  
  inflating: Restore.plist           
  adding: 058-74940-063.dmg (deflated 17%)
  adding: 058-75094-062.dmg (deflated 1%)
  adding: 058-75381-062.dmg (deflated 17%)
  adding: BuildManifest.plist (deflated 95%)
  adding: Firmware/ (stored 0%)
  adding: Firmware/all_flash/ (stored 0%)
  adding: Firmware/all_flash/applelogo@2x~ipad.s5l8960x.im4p (deflated 25%)
  adding: Firmware/all_flash/applelogo@2x~ipad.s5l8960x.im4p.plist (deflated 38%)
  adding: Firmware/all_flash/batterycharging0@2x~ipad.s5l8960x.im4p (deflated 19%)
  adding: Firmware/all_flash/batterycharging0@2x~ipad.s5l8960x.im4p.plist (deflated 27%)
  adding: Firmware/all_flash/batterycharging1@2x~ipad.s5l8960x.im4p (deflated 12%)
  adding: Firmware/all_flash/batterycharging1@2x~ipad.s5l8960x.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/batteryfull@2x~ipad.s5l8960x.im4p (deflated 83%)
  adding: Firmware/all_flash/batteryfull@2x~ipad.s5l8960x.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/batterylow0@2x~ipad.s5l8960x.im4p (deflated 83%)
  adding: Firmware/all_flash/batterylow0@2x~ipad.s5l8960x.im4p.plist (deflated 25%)
  adding: Firmware/all_flash/batterylow1@2x~ipad.s5l8960x.im4p (deflated 19%)
  adding: Firmware/all_flash/batterylow1@2x~ipad.s5l8960x.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/DeviceTree.j71ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j71ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/DeviceTree.j72ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j72ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/DeviceTree.j73ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j73ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/DeviceTree.j85ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j85ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/DeviceTree.j86ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j86ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/DeviceTree.j87ap.im4p (deflated 83%)
  adding: Firmware/all_flash/DeviceTree.j87ap.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/glyphplugin@2x~ipad-lightning.s5l8960x.im4p (deflated 14%)
  adding: Firmware/all_flash/glyphplugin@2x~ipad-lightning.s5l8960x.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/iBoot.ipad4.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/iBoot.ipad4.RELEASE.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/iBoot.ipad4b.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/iBoot.ipad4b.RELEASE.im4p.plist (deflated 28%)
  adding: Firmware/all_flash/LLB.ipad4.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/LLB.ipad4.RELEASE.im4p.plist (deflated 25%)
  adding: Firmware/all_flash/LLB.ipad4b.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/LLB.ipad4b.RELEASE.im4p.plist (deflated 27%)
  adding: Firmware/all_flash/recoverymode@2x~ipad-lightning.s5l8960x.im4p (deflated 83%)
  adding: Firmware/all_flash/recoverymode@2x~ipad-lightning.s5l8960x.im4p.plist (deflated 26%)
  adding: Firmware/all_flash/sep-firmware.j71.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j71.RELEASE.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/sep-firmware.j72.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j72.RELEASE.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/sep-firmware.j73.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j73.RELEASE.im4p.plist (deflated 38%)
  adding: Firmware/all_flash/sep-firmware.j85.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j85.RELEASE.im4p.plist (deflated 37%)
  adding: Firmware/all_flash/sep-firmware.j86.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j86.RELEASE.im4p.plist (deflated 38%)
  adding: Firmware/all_flash/sep-firmware.j87.RELEASE.im4p (deflated 0%)
  adding: Firmware/all_flash/sep-firmware.j87.RELEASE.im4p.plist (deflated 37%)
  adding: Firmware/dfu/ (stored 0%)
  adding: Firmware/dfu/iBEC.ipad4.RELEASE.im4p (deflated 0%)
  adding: Firmware/dfu/iBEC.ipad4.RELEASE.im4p.plist (deflated 27%)
  adding: Firmware/dfu/iBEC.ipad4b.RELEASE.im4p (deflated 50%)
  adding: Firmware/dfu/iBEC.ipad4b.RELEASE.im4p.plist (deflated 26%)
  adding: Firmware/dfu/iBSS.ipad4.RELEASE.im4p (deflated 0%)
  adding: Firmware/dfu/iBSS.ipad4.RELEASE.im4p.plist (deflated 26%)
  adding: Firmware/dfu/iBSS.ipad4b.RELEASE.im4p (deflated 52%)
  adding: Firmware/dfu/iBSS.ipad4b.RELEASE.im4p.plist (deflated 26%)
  adding: Firmware/Mav7Mav8-7.60.00.Release.bbfw (deflated 0%)
  adding: Firmware/Mav7Mav8-7.60.00.Release.plist (deflated 29%)
  adding: Firmware/usr/ (stored 0%)
  adding: Firmware/usr/local/ (stored 0%)
  adding: Firmware/usr/local/standalone/ (stored 0%)
  adding: Restore.plist (deflated 86%)
  adding: kernelcache.release.ipad4 (deflated 6%)
  adding: kernelcache.release.ipad4b (deflated 6%)
rm: it_worked: No such file or directory
rm: ipsw/: is a directory
Your current ApNonce is ee4b7f9b2d7d41bfde4c8390734a83d63c2fe997
7758956dff0
Your device is an iPad4,4
Your boardconfig is j85ap
blankfile upload
[==================================================] 100.0%
ibss upload
[==================================================] 100.0%
ibec upload
dyld: Symbol not found: _clonefile
  Referenced from: /Applications/LeetDown.app/Contents/rsr/./bin/futurerestore
  Expected in: /usr/lib/libSystem.B.dylib
 in /Applications/LeetDown.app/Contents/rsr/./bin/futurerestore
/Applications/LeetDown.app/Contents/rsr/down.sh: line 121: 44228 Trace/BPT trap: 5       ./bin/futurerestore -t shsh.shsh2 -p BM/iPad4,4.plist -s sep.im4p -m BM/iPad4,4.plist custom.ipsw --no-baseband
fail
rm: ipsw_create_err: No such file or directory
rm: libusb_missing: No such file or directory
rm: no_device: No such file or directory
rm: no_dfu: No such file or directory
rm: *.patched: No such file or directory
rm: *.shsh: No such file or directory
rm: *.bbfw: No such file or directory
![image](https://user-images.githubusercontent.com/83359255/126730036-a86b34e8-fac7-428d-a960-8b39c695274b.png)

**Comment by PENG-OX** [comment]
ipad mini 2  ios 12.5.1

**Comment by PENG-OX** [comment]
I succeeded 
Share the steps to success below
My device is ipad mini 2 ios 12.5.4 and my Apple computer is 2015 MBP
I tried to downgrade on osx 10.11 10.12（Tried more than 50 times） 10.13 using leetDown 1.0.5 and finally succeeded on 10.13
1. Install python3 (mine is 3.9.6)
2. Before starting leetDown, use "sudo killall -STOP -c usbd" to input several times in succession until it does not prompt for password input
3. Execute in the order of leetDown operation. I tried 6 times and failed for the seventh time.
<img width="1105" alt="成功" src="https://user-images.githubusercontent.com/83359255/126789395-c5bec8df-128e-4fd9-8eba-58e102a8baa2.png">
